### PR TITLE
Remove Internet Explorer specific code

### LINKED ISSUE #886: Remove Internet Explorer specific code
Internet Explorer has officially retired and out of support. Given that our `.browserslistrc` is already set to `not dead`. It is not longer necessary to include Internet Explorer specific code in codebase

As per [Browserslist documentation](https://browsersl.ist/#q=dead) IE considered 'dead'

Here are the proposed changes:

* Clean IE from `.browserslistrc` file 
* Remove browser specific checks like `isIE`, `documentMode`, `MSInpiutMethodContext`

### PR DESCRIPTION
**Pull request by gulderov**
### Description
This pull request cleans the codebase from IE-specific code as described in #886 

### Issues Resolved
closes #886

### Check List
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
 ### 🛠 Maintenance
 
+- Remove Internet Explorer specific code (#890)
```
Comment (BSFishy) [inline CHANGELOG.md:27]: Also, could this be a link to the PR?

```suggestion
- Remove Internet Explorer specific code ([#890](https://github.com/opensearch-project/oui/pull/890))
```

Comment (gulderov) [reply CHANGELOG.md]: I have include link to the pull request

File: src-docs/src/components/guide_section/guide_section.tsx [hunk 2]
```diff
@@ -132,14 +132,7 @@ export const GuideSection: FunctionComponent<GuideSection> = ({
   };
 
   const renderPlaygroundToggle = () => {
-    const isPlaygroundUnsupported =
-      typeof window !== 'undefined' &&
-      typeof document !== 'undefined' &&
-      !!window.MSInputMethodContext &&
-      // @ts-ignore doesn't exist?
-      !!document.documentMode;
```
[the remark this answers is not in the document]
Comment (gulderov) [reply src-docs/src/components/guide_section/guide_section.tsx]: That is an excellent point. I have made the adjustments

File: packages/react-datepicker.js [hunk 3]
```diff
@@ -6008,7 +5973,7 @@ function getOffsetRectRelativeToArbitraryNode(children, parent) {
     offsets.marginLeft = marginLeft;
   }
 
-  if (isIE10 && !fixedPosition ? parent.contains(scrollParent) : parent === scrollParent && scrollParent.nodeName !== 'BODY') {
```
Comment (BSFishy) [inline packages/react-datepicker.js:6011]: What is the order of operations here? Is it `(isIE10 && !fixedPosition) ? ...` or `isIE10 && (!fixedPosition ? ...)`? I think that determines how to rewrite this

Comment (gulderov) [reply packages/react-datepicker.js:6011]: [Precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence) is 4 for logical AND vs 2 for ternary operator

It is possible to put braces like this: 
`if ((isIE10 && !fixedPosition) ? parent.contains(scrollParent) : parent === scrollParent && scrollParent.nodeName !== 'BODY')`

As `isIE10` is always `false` we could substitute with
`if ((false && !fixedPosition) ? parent.contains(scrollParent) : parent === scrollParent && scrollParent.nodeName !== 'BODY')`

In this case `false && !fixedPosition` will always evaluate to `false`. No matter the value of `fixedPosition`. So the entire expression will evaluate to what is after colon

**Review by BSFishy — APPROVED** [review]

**Review by AMoo-Miki — APPROVED** [review]
Thanks for this change. I would love to revisit `browserlist` and get t to match that of OSD's.

File: CHANGELOG.md [hunk 4]
```diff
@@ -72,6 +72,7 @@
 
 - [CVE-2023-26136] Add resolution for tough-cookie to ^4.1.3 ([#889](https://github.com/opensearch-project/oui/pull/889))
 - [CVE-2023-26115] Bump word-wrap from 1.2.3 to 1.2.4 ([#891](https://github.com/opensearch-project/oui/pull/891))
+- Remove Internet Explorer specific code ([#890](https://github.com/opensearch-project/oui/pull/890))
```
Comment (BSFishy) [inline CHANGELOG.md:75]: This should be moved to `unreleased` section before merging. This is an artifact of how updating the changelog was done, sorry!

**Review by BSFishy — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (gulderov) [reply CHANGELOG.md]: done

**Review by BSFishy — APPROVED** [review]
Perfect!

**Pull request merged**